---
title: "CITEseq_preprocessing"
output: html_document
date: "2025-11-17"
author: Marissa Esteban
---

RMD that loads and preprocesses (preprocessing and cell type annotation) CITEseq data
NOTE: *.loom splicing matrix is created from Velocyto.

data: YS006_UW

At the time of creating this rmd, I've only generated the .loom file for YS006.

```{r}
knitr::opts_chunk$set(echo = TRUE)

# Set working directory for chunk outputs
knitr::opts_knit$set(root.dir = "/Users/marissaestaban/Documents/CITEseq_data")

# Install needed packages
# install.packages("Seurat")
# install.packages("SeuratObject")
# install.packages("dplyr")
# install.packages("patchwork")
# install.packages("ggplot2")
# install.packages('BiocManager')
# BiocManager::install('glmGamPoi')

# Load packages needed for analysis
library(Seurat)
library(dplyr)
library(patchwork)
library(ggplot2)
```

Loading the data and coversion into Seurat objects

```{r}
#Ensures the directory is set to the folder containing the unpacked data files

# loading UW
cite_uw <- Read10X("/Users/marissaestaban/Documents/CITEseq_data/YS006_UW/filtered_feature_bc_matrix")
```

To handle the cite-seq datasets, the modifiers attached to the ADT data must be removed
```{r}

#ncontrols dictates the number of highly expressed genes from the control group to be kept - for practice the control group is human genes, for our actual data it is Isotype
#Because some of the TotalSeqB antibodies are against both human, rat, and mouse proteins, need to remove Hu and Rt prefix as well
cite_uw[["Antibody Capture"]] <- CollapseSpeciesExpressionMatrix(cite_uw[["Antibody Capture"]], prefix = "HuMsRt.", controls = "Isotype", ncontrols = 25)
cite_uw[["Antibody Capture"]] <- CollapseSpeciesExpressionMatrix(cite_uw[["Antibody Capture"]], prefix = "HuMs.", controls = "Isotype", ncontrols = 25)
cite_uw[["Antibody Capture"]] <- CollapseSpeciesExpressionMatrix(cite_uw[["Antibody Capture"]], prefix = "MsRt.", controls = "Isotype", ncontrols = 25)
cite_uw[["Antibody Capture"]] <- CollapseSpeciesExpressionMatrix(cite_uw[["Antibody Capture"]], prefix = "Ms.", controls = "Isotype", ncontrols = 25)
```

Dropping the protein data:
We do not need the protein data for RNA velocity so we will drop this assay for now.

```{r}
# converting into Seurat objects
uw <- CreateSeuratObject(counts = cite_uw[["Gene Expression"]], min.cells = 3, project = "CITEseq_YS006")
uw[["ADT"]] <- CreateAssayObject(cite_uw[["Antibody Capture"]][, colnames(x = uw)])


# adding timepoints to object
uw$timepoint <- "Unwounded"
```

Data cleaning

```{r}

#Mitochondrial transcript percentage
uw[["percent_mito"]] <- PercentageFeatureSet(uw, pattern = "^mt-")

#Ribosomal transcript percentage
uw[["percent_ribo"]] <- PercentageFeatureSet(uw, pattern = "^Rp[sl]")

#Violin plot of QC metrics, specifying number of columns (ncol) and removing the x-axis title
VlnPlot(uw, features = c('nFeature_RNA','nCount_RNA','percent_mito','percent_ribo'), ncol = 2) & theme(axis.title.x = element_blank())
```

```{r}
plot1 <- FeatureScatter(uw, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
plot2 <- FeatureScatter(uw, feature1 = "nCount_RNA", feature2 = "percent_mito")
plot3 <- FeatureScatter(uw, feature1 = "nCount_RNA", feature2 = "percent_ribo")
plot4 <- FeatureScatter(uw, feature1 = "percent_ribo", feature2 = "percent_mito")

plot1 + plot2 + plot3 + plot4 + plot_layout(guides = "collect")
```
Removing low quality cells

```{r}
# subset data now
uw <- subset(uw, subset = nFeature_RNA > 200 & nFeature_RNA < 5000 & percent_mito < 10)

ys006 <- subset(uw, subset = orig.ident == "CITEseq_YS006")
```

```{r}
# checking
plot1 <- FeatureScatter(uw, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
plot2 <- FeatureScatter(uw, feature1 = "nCount_RNA", feature2 = "percent_mito")
plot3 <- FeatureScatter(uw, feature1 = "nCount_RNA", feature2 = "percent_ribo")
plot4 <- FeatureScatter(uw, feature1 = "percent_ribo", feature2 = "percent_mito")

plot1 + plot2 + plot3 + plot4 + plot_layout(guides = "collect")
```
Data Normalization

```{r}
ys006 <- SCTransform(ys006, vars.to.regress = "percent_mito")
```

```{r}
ys006 <- RunPCA(ys006)
```


```{r}
DimHeatmap(ys006, dims = 1:9, cells = 500, balanced = TRUE)
ElbowPlot(ys006, ndims = 40)
```


```{r High PC DimHeatmap}
DimHeatmap(ys006, dims = 15:20, cells = 500, balanced = TRUE)
```



```{r YS006: Find neighbors, clusters, and run UMAP}
ys006 <- ys006 %>%
  FindNeighbors(reduction = "pca", dims = 1:17, verbose = FALSE) %>%
  FindClusters(resolution = 1, verbose = FALSE) %>%
  RunUMAP(reduction = "pca", dims = 1:17, verbose = FALSE)

DimPlot(ys006)
```



```{r Feature plots}
FeaturePlot(ys006, c("Krt14", "Pdgfra","Ptprc","Pecam1"))
```

In addition to plotting various meta data, we can also plot the expression of marker genes overlayed on the UMAP in what is called a FeaturePlot. This allows direct visualization of gene expression and allows us to distinguish major clusters visually. Here, we plot the expression of keratinocyte, fibroblast, immune cell, and endothelial markers by supplying a vector of genes to plot.

```{r Save data}
# setwd("/Users/marissaestaban/Documents/CITEseq_data")
# saveRDS(ys006, "CITEseq_discovery_YS006.rds")
```


CLUSTERING AND ANNOTATION: YS006 (UW)

```{r}
# ys006.sct <- PrepSCTFindMarkers(ys006, assay = "SCT", verbose = TRUE)
# ys006.sct <- readRDS("/Users/marissaestaban/Documents/CITEseq_data/ys006_sct.rds")
markers = FindAllMarkers(ys006, only.pos = T, min.pct = .25)

top_markers <- markers %>%
  group_by(cluster) %>%
  slice_head(n = 3)


p <- DotPlot(ys006, unique(top_markers$gene), dot.scale = 7, cols =  c("white","orangered3")) +
  RotatedAxis() +
  theme(
    text = element_text(size = 12, face = "bold", family = "Arial")
  )
p
```


```{r}
features = c("Krt14","Itga6", #Keratincoytes IFE 
             "Krt15","Sox9", #Keratinocytes HF 
             "Pparg", #Sebocytes
             "Col1a2","Pdgfra", #Fibroblast 
             "Ptprc", # Immune Cell 
             "Pecam1", #Endothelial Cell 
              "Rgs5", #Pericytes
             "Mylpf", "Des", # Muscle 
              "Mpz", #Schwann Cell 
             "Mlana", #Melanocyte 
             "Adipoq", #Adipocyte 
             "Hba-a1","Hbb-bt" #RBC 
)

expressed_features <- features[features %in% rownames(ys006)]
VlnPlot(ys006, expressed_features, stack = T, flip =T)
```

```{r}
new.cluster.ids <- c("Keratinocytes IFE", #0
                     "Endothelial Cells", #1
                     "Fibroblasts", #2
                     "Immune Cells", #3
                     "Immune Cells", #4
                     "Immune Cells", #5
                     "Fibroblasts", #6
                     "Muscle", #7
                     "Keratinocytes IFE", #8
                     "Immune Cells", #9
                     "Keratinocytes IFE", #10
                     "Immune Cells", #11
                     "Keratinocytes HF", #12
                     "Fibroblast", #13
                     "Muscle", #14
                     "Keratinocytes IFE", #15
                     "Schwann Cells", #16
                     "Immune Cells" #17
)                     
                     
                     
                     
names(new.cluster.ids) <- levels(ys006)
ys006_named <- RenameIdents(ys006, new.cluster.ids)
```

Plotting with new cluster names

```{r}
DimPlot(ys006_named, reduction = "umap")
levels(ys006_named)
```
```{r}
ys006_named$celltype <- Idents(ys006_named)
```


```{r Save annotated data}

# Normalize RNA last to the RNA layer
DefaultAssay(ys006_named) <- "RNA"
ys006_named <- NormalizeData(ys006_named)


setwd("/Users/marissaestaban/Documents/CITEseq_data")
saveRDS(ys006_named, "ys006_named")
```

```{r}

```

